system/src/system_task.cpp

```diff
@@ -467,6 +467,26 @@ void manage_ble_prov_mode() {
 #endif
 }
 
+extern "C" void system_isr_task_queue_free_memory(uint8_t *ptrToFree) {
+    struct FreeTask : ISRTaskQueue::Task {
+        uint8_t *arg;
+    };
+
+    auto task = static_cast<FreeTask*>(system_pool_alloc(sizeof(FreeTask), nullptr));
+    if (!task)
+    {
+        return;
```

Comment:
This will result in a memory leak, but I think we have no other recourse here?

Reply:
Make this return `int` and return `SYSTEM_ERROR_NO_MEMORY`. Handle it in `_freertos_mfree()` or more precisely its strong override in rtl_sdk_support.cpp and SPARK_ASSERT.